data/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/Conversation.kt

```diff
@@ -172,6 +172,7 @@ data class Conversation(
         sealed class Group : Type {
             data object Regular : Group()
             data object Channel : Group()
+            companion object : Group()
```

Comment:
Is this intended?

This adds a third "type":

```kotlin
val regularGroup = Conversation(type = Group.Regular)
val channel = Conversation(type = Group.Channel)
val whatIsThis = Conversation(type = Group) // What is this???
```

Also makes this possible:

```kotlin
when(type){
    Group -> {} // What is this case?
    Group.Regular -> {}
    Group.Channel -> {}
}
```

Reply:
It is a type that can be a channel or regular group. 
Added because, in `ConversationEntity`, we have a type `GROUP` that we don't know what it is without fetching db for `isChannel` value.
Mainly added for calling, currently we pass `RegularGroup` for it, which is confusing and I don't think it is worth it to check for `isChannel` value every time. That why `Group` companion object  is added.

I think a cleaner solution would be to add a channel type to ConversationEntity instead of having a separate  boolean value for it.

Reply:
For Calling, I guess there is zero distinction between channel or regular groups, so I understand that we would need to pass a generic "Group" to Calling. But this is still odd. It breaks the typing possibilities, adding something else that isn't a channel, nor a regular group. This can lead to weird bugs.

Where is this being passed to Calling? Calling should check `is Group` instead of `== Group`.

Whenever we fetch the data from the Database, we also have access to `isChannel`, so we can already figure out the type.

Reply:
Maybe with a clearer example of what I mean:

```kotlin
val regularGroup = Conversation(type = Group.Regular)
val channel = Conversation(type = Group.Channel)
val whatIsThis = Conversation(type = Group) // What is this???


assertTrue { regularGroup.type is Group } // true
assertTrue { regularGroup.type == Group } // FALSE

assertTrue { channel.type is Group } // true
assertTrue { channel.type == Group } // FALSE
```

Reply:
Removed it